src/NUnitFramework/framework/Constraints/PropertyConstraint.cs

```diff
@@ -56,21 +57,33 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             // TODO: Use an error result for null
             Guard.ArgumentNotNull(actual, nameof(actual));
+            const BindingFlags bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance;
 
-            Type actualType = actual as Type;
-            if (actualType == null)
-                actualType = actual.GetType();
+            PropertyInfo property = Reflect.GetUltimateShadowingProperty(typeof(TActual), name, bindingFlags);
 
-            PropertyInfo property = Reflect.GetUltimateShadowingProperty(actualType, name,
-                BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            if (property == null && typeof(TActual).IsInterface && typeof(TActual).GetInterfaces() is { } interfaces && interfaces.Length > 0)
```

Comment:
This null check can be removed because Type.GetInterfaces() is documented to return an empty array rather than null, following the general pattern.
```suggestion
            if (property == null && typeof(TActual).IsInterface && typeof(TActual).GetInterfaces() is var interfaces && interfaces.Length > 0)
```

Reply:
I actually see the length check as redundant with the foreach loop, so I'd loop over `typeof(TActual).GetInterfaces()` directly.

Reply:
sure, done. I'd be cautious with MS declarations especially given you're probably compiling against multiple legacy targets, but I assume this time we can trust them ;-)